### PR TITLE
Image to image

### LINKED ISSUE #364: Text-to-Image Task Page
This is the tracker for completion of task page for `text-to-image` task. The page is here: https://huggingface.co/tasks/text-to-image You can contribute parts of a task or the whole task! (There's no single assignee to a task, these are wikipedia-like pages)
You can refer to https://github.com/huggingface/hub-docs/issues/369 if you want to contribute! 🤗

List of items to contribute:

- [x] Task description
- [x] Subtasks (if there is any)
- [x] Most used dataset for the task
- [x] Most used model for the task
- [x] Metrics that are used to evaluate the task
- [x] Libraries used for the task
- [x] Use cases (you can still add more!)
- [x] Useful Resources (you can still add more!)
- [x] Small snippet for inference that demonstrates the task
- [x] Task Schema
- [x] Model for the Widget

### PR DESCRIPTION
**Pull request by Paul92**
Fixes #364.

As a first PR, I'd be grateful for some comments regarding the style and nature of the introductory content.

I plan to address a few of the remaining tasks (example code, example input/output, etc) in the meantime.

### CODE REVIEW

File: tasks/src/image-to-image/data.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { type TaskDataCustom } from "../Types";
+
+const taskData: TaskDataCustom = {
```
Comment (merveenoyan) [inline tasks/src/image-to-image/data.ts:3]: I think it makes sense to have only VIDIT and CelebA datasets, we look for datasets with very general use cases 🙂

File: tasks/src/image-to-image/data.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		outputs: [],
+	},
+	isPlaceholder: false,
+	metrics:       [
+		{
+			description: "Peak Signal to Noise Ratio (PSNR) is an approximation of the human perception, considering the ratio of the absolute intensity with respect to the variations. Measured in dB, a high value indicates a high fidelity.",
+			id:          "PSNR"
+		},
+		{
+			description: "Structural Similarity Index (SSIM) is a perceptual metric which compares the luminance, constrast and structure of two images. Between -1 and 1, the higer values indicate closer resemblance.",
```
Comment (merveenoyan) [inline tasks/src/image-to-image/data.ts:33]:
```suggestion
			description: "Structural Similarity Index (SSIM) is a perceptual metric which compares the luminance, constrast and structure of two images. The values of SSIM range between -1 and 1, and higher values indicate closer resemblance to the original image.",
```

File: tasks/src/image-to-image/data.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+			id:          "SSIM"
+		},
+		{
+			description: "Inception Score (IS) is an analysis of the labels predicted by an image classification model when presented with a sample of the generated images.",
```
Comment (merveenoyan) [inline tasks/src/image-to-image/data.ts:37]: How can we use inception score in this task?

File: tasks/src/image-to-image/data.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+			id:          "IS"
+		},
+	],
+	models:        [
+		{
+			description: "Enhances images captures in low light conditions",
```
Comment (merveenoyan) [inline tasks/src/image-to-image/data.ts:43]:
```suggestion
			description: "A model that enhances images captured in low light conditions.",
```

File: tasks/src/image-to-image/data.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+			id: "keras-io/low-light-image-enhancement"
+		}
+		{
+			description: "Increases the resolution of an image",
```
Comment (merveenoyan) [inline tasks/src/image-to-image/data.ts:47]:
```suggestion
			description: "A model that increases the resolution of an image.",
```

File: tasks/src/image-to-image/data.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+			id: "keras-io/super-resolution"
+		}
+		{
+			description: "Transforms photographies into a Japanese animation style image",
+			id: "akiyamasho/AnimeBackgroundGAN-Shinkai"
+		}
+	],
+	summary:       "Image to image models aim at transforming a source image to match the characteristics of a target image or a target image domain. There is an endless list of image manipulations and enhancements possible using image to image machine learning models."
```
Comment (merveenoyan) [inline tasks/src/image-to-image/data.ts:55]:
```suggestion
	summary:       "Image to image is the task of transforming a source image to match the characteristics of a target image or a target image domain. Any image manipulation and enhancement is possible with image to image models."
```
I loved the definition, just leaving few nits ❤️

File: tasks/src/image-to-image/about.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+style of Claude Monet.
+
+### Image segementation
+
+Image segmentation represents is a pixel level classification - each pixel is assigned a numeric label, according to
+the object it belongs to.
+
+### Image inpainting
+
+Image inpaining is widely used during photography editing to remove unwanted objects, such as poles, wires or sensor
```
Comment (merveenoyan) [inline tasks/src/image-to-image/about.md:15]: I'm a bit inclined to have Image Inpainting as a sub-task in task variants, what do you think? 🙂 I feel like other use cases are general enough to be in task variants as well, except for Style Transfer.

File: tasks/src/image-to-image/about.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+## Use Cases
+
+### Style transfer
+
+One of the first and most popular applications, style transfer can convert a regular photography into a paining in the
```
Comment (merveenoyan) [inline tasks/src/image-to-image/about.md:5]:
```suggestion
One of the most popular use cases of image to image is the style transfer. Style transfer models can convert a regular photography into a painting in the style of a famous painter.
```

**Review by merveenoyan — COMMENTED** [review]
Looks great to me, thanks for handling this! ❤️ I left minor nits 🙂

File: tasks/src/image-to-image/data.ts (hunk 3 above)
Comment (Paul92) [reply tasks/src/image-to-image/data.ts:37]: IS can be used (with ups and downs) to evaluate the output of generative models by analyzing the distribution of the output. Very intuitively, if we want to make image to image translation of a tree, we can run the results through a classification model to make sure we still have a tree, and asses the confidence of the classifier for the quality of the image.

https://arxiv.org/pdf/1801.01973.pdf
https://arxiv.org/pdf/2101.08629.pdf (section II c)

File: tasks/src/image-to-image/about.md (hunk 7 above)
Comment (Paul92) [reply tasks/src/image-to-image/about.md:15]: Considering the broadness of image to image translation, we can have many interpretations of what is a variant, and what really is the core application :). But I agree all are very broad applications.

**Comment by Paul92** [comment]
I've been thinking what to share as a code example. Perhaps something along the lines of this low level enhancement app?

https://huggingface.co/spaces/keras-io/low-light-image-enhancement/blob/main/app.py

File: tasks/src/image-to-image/about.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+### Style transfer
+
+One of the most popular use cases of image to image is the style transfer. Style transfer models can convert a regular photography into a painting in the style of a famous painter.
+style of Claude Monet.
+
+## Task Variants
+
+### Image inpainting
+
+Image inpaining is widely used during photography editing to remove unwanted objects, such as poles, wires or sensor
```
Comment (merveenoyan) [inline tasks/src/image-to-image/about.md:12]:
```suggestion
Image inpainting is widely used during photography editing to remove unwanted objects, such as poles, wires or sensor
```
nit :)

File: tasks/src/image-to-image/data.ts [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+			description: "A model that enhances images captured in low light conditions.",
+			id: "keras-io/low-light-image-enhancement"
+		}
+		{
+			description: "A model that increases the resolution of an image.",
+			id: "keras-io/super-resolution"
+		}
+	],
+	summary:       "Image to image is the task of transforming a source image to match the characteristics of a target image or a target image domain. Any image manipulation and enhancement is possible with image to image models."
+	widgetModels:  ["keras-io/low-light-image-enhancement"],
```
Comment (merveenoyan) [inline tasks/src/image-to-image/data.ts:54]: This model's widget is unfortunately not available (needs to be implemented) and from what I checked, there's no models with widget at the moment so we can remove it 🙂.
```suggestion
	widgetModels:  [""],
```

File: tasks/src/image-to-image/data.ts [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+		{
+			description: "Structural Similarity Index (SSIM) is a perceptual metric which compares the luminance, constrast and structure of two images. The values of SSIM range between -1 and 1, and higher values indicate closer resemblance to the original image.",
+			id:          "SSIM"
+		},
+		{
+			description: "Inception Score (IS) is an analysis of the labels predicted by an image classification model when presented with a sample of the generated images.",
+			id:          "IS"
+		},
+	],
+	models:        [
```
Comment (merveenoyan) [inline tasks/src/image-to-image/data.ts:43]: On top of these models, maybe we could add https://huggingface.co/lambdalabs/sd-image-variations-diffusers what do you think?

File: tasks/src/image-to-image/data.ts [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+		]
+	},
+	isPlaceholder: false,
+	metrics:       [
+		{
+			description: "Peak Signal to Noise Ratio (PSNR) is an approximation of the human perception, considering the ratio of the absolute intensity with respect to the variations. Measured in dB, a high value indicates a high fidelity.",
+			id:          "PSNR"
+		},
+		{
+			description: "Structural Similarity Index (SSIM) is a perceptual metric which compares the luminance, constrast and structure of two images. The values of SSIM range between -1 and 1, and higher values indicate closer resemblance to the original image.",
```
Comment (merveenoyan) [inline tasks/src/image-to-image/data.ts:35]:
```suggestion
			description: "Structural Similarity Index (SSIM) is a perceptual metric which compares the luminance, contrast and structure of two images. The values of SSIM range between -1 and 1, and higher values indicate closer resemblance to the original image.",
```
nit 🙂

File: tasks/src/image-to-image/about.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
+## Use Cases
+
+### Style transfer
+
+One of the most popular use cases of image to image is the style transfer. Style transfer models can convert a regular photography into a painting in the style of a famous painter.
+style of Claude Monet.
```
Comment (merveenoyan) [inline tasks/src/image-to-image/about.md:6]:
```suggestion
```
We can remove this given we fixed above paragraph 🙂

File: tasks/src/image-to-image/data.ts [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+		},
+		{
+			description: "Multiple images of celebrities, used for facial expression translation"
+			id: "CelebA"
+		}
+	],
+	demo:     {
+		inputs: [
+			{
+				filename: "image-to-image-input.jpeg",
```
Comment (merveenoyan) [inline tasks/src/image-to-image/data.ts:17]: ❤️
Ekran Resmi 2022-10-07 17 34 02

**Review by merveenoyan — COMMENTED** [review]
LGTM! @osanseviero can you check super quick?

**Review by Paul92 — COMMENTED** [review]
Added lambdalabs/sd-image-variations-diffusers

File: tasks/src/image-to-image/data.ts [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+
+const taskData: TaskDataCustom = {
+	datasets: [
+		{
+			description: "Synthetic dataset, for image relighting",
+			id: "VIDIT"
+		},
+		{
+			description: "Multiple images of celebrities, used for facial expression translation"
+			id: "CelebA"
```
Comment (osanseviero) [inline tasks/src/image-to-image/data.ts:11]: You can use this one from the Hub
```suggestion
			id: "huggan/CelebA-faces"
```

File: tasks/src/image-to-image/data.ts [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+		}
+	],
+	demo:     {
+		inputs: [
+			{
+				filename: "image-to-image-input.jpeg",
```
Comment (osanseviero) [inline tasks/src/image-to-image/data.ts:17]: Very cool!

**Review by osanseviero — APPROVED** [review]
Very cool, thanks!

**Comment by merveenoyan** [comment]
@Paul92 if you commit suggestion of osanseviero above I'm happy to merge your PR 🙂

**Comment by merveenoyan** [comment]
@Paul92 just merged. Can you give me your hugging face handle so I can add it at the end of the page to give you credits? 🙂

File: tasks/src/image-to-image/data.ts (hunk 14 above)
Comment (Joggyjagz7) [reply tasks/src/image-to-image/data.ts:17]: Is this repo still open for contribution?
Awesome work so far

**Pull request merged**